microsoft/testsuites/hpc/infinibandsuit.py

```diff
@@ -249,3 +249,97 @@ def verify_intel_mpi(self, environment: Environment, log: Logger) -> None:
+            ]
+        )
+
+        server_ib = server_node.features[Infiniband]
+        client_ib = client_node.features[Infiniband]
+        run_in_parallel([server_ib.install_open_mpi, client_ib.install_open_mpi])
+
+        # Restart the ssh sessions for changes to /etc/security/limits.conf
+        # to take effect
+        server_node.close()
+        client_node.close()
+
+        # Get the ip adresses and device name of ib device
+        server_ib_interfaces = server_ib.get_ib_interfaces()
+        client_ib_interfaces = client_ib.get_ib_interfaces()
+        server_ip = server_ib_interfaces[0].ip_addr
+        client_ip = client_ib_interfaces[0].ip_addr
+
+        # Test relies on machines being able to ssh into each other
+        server_ssh = server_node.tools[Ssh]
+        client_ssh = client_node.tools[Ssh]
+        server_ssh.enable_public_key(client_ssh.generate_key_pairs())
+        client_ssh.enable_public_key(server_ssh.generate_key_pairs())
+        server_ssh.add_known_host(client_ip)
+        client_ssh.add_known_host(server_ip)
+
+        find = server_node.tools[Find]
+        findResults = find.find_files(
+            server_node.get_pure_path("/"), "IMB-MPI1", sudo=True
+        )
+        assert_that(len(findResults)).described_as(
+            "Could not find location of IMB-MPI1 for Open MPI"
+        ).is_greater_than(0)
+        test_path = findResults[0]
+        server_node.execute(
+            f"/usr/local/bin/mpirun --host {server_ip},{server_ip} "
+            "-n 2 --mca btl self,vader,openib --mca btl_openib_cq_size 4096 "
+            "--mca btl_openib_allow_ib 1 --mca "
+            f"btl_openib_warn_no_device_params_found 0 {test_path} pingpong",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="Failed intra-node ping pong test "
+            "with Open MPI",
+        )
+
+        tests = {
+            "IMB-MPI1": "",
+            "IMB-RMA": " One_put_all One_get_all "
+            "All_put_all All_get_all Exchange_put Exchange_get",
+            "IMB-P2P": "",
+        }
+
+        for test_name, extra_params in tests.items():
+            findResults = find.find_files(
+                server_node.get_pure_path("/"), test_name, sudo=True
+            )
+            assert_that(len(findResults)).described_as(
+                f"Could not find location of Open MPI test: {test_name}"
+            ).is_greater_than(0)
+            test_path = findResults[0]
+            server_node.execute(
+                f"/usr/local/bin/mpirun --host {server_ip},{client_ip} "
+                "-n 2 --mca btl self,vader,openib --mca btl_openib_cq_size 4096 "
+                "--mca btl_openib_allow_ib 1 --mca "
+                f"btl_openib_warn_no_device_params_found 0 {test_path}{extra_params}",
+                expected_exit_code=0,
+                expected_exit_code_failure_message="Failed "
+                f"{test_name} test with Open MPI",
+                timeout=1200,
```

Comment:
How long this test case will run? We should make sure it's as short as possible.